docs/extend-stages.md

```diff
@@ -1,25 +1,31 @@
 # Extension Life cycle
 
+This document describes the process of publishing extensions within the CiviCRM
+ecosystem.
+
+## Consumer vs. developer roles
```

Comment:
Overall, I like the changes. 

For a shorter title, maybe just "Background"? That's the title which usually means "Skip past this if your skimming for substance".